### ISSUE TITLE
RFE: Figure out a way to parse the Gitlab API docs so it can be used to validate we support everything

### ISSUE THREAD

**Issue by JohnVillalovos** [python-gitlab/python-gitlab#1446]
Putting this here so I don't forget.

Would be nice if we could parse the API documentation (which is in Gitlab Markdown format) and extract the API data so that we could use that to figure out if we are missing anything in our implementation.

A possibility is something like Pandoc https://pandoc.org/ or just write some code to try to do it.

**Comment by max-wittig** [comment]
Not sure if that makes sense. We could do it once an `openapi.yaml` for GitLab is available. They are still in the early stages:
* https://gitlab.com/gitlab-org/gitlab/-/issues/16023

**Comment by nejch** [comment]
> 
> 
> Not sure if that makes sense. We could do it once an `openapi.yaml` for GitLab is available. They are still in the early stages:
> 
>     * https://gitlab.com/gitlab-org/gitlab/-/issues/16023

Or use that approach to help draft/convert the existing GitLab docs to openapi.yml for more endpoints 🤔 

It's pretty quiet in there yeah:
https://gitlab.com/gitlab-org/gitlab/-/tree/master/doc/api/openapi/v4

At least some very basic stuff could be pre-filled into yaml for endpoints that follow the standard API doc guideline:
https://docs.gitlab.com/ee/development/documentation/restful_api_styleguide.html#api-topic-template

Just thinking out loud also, since gitlab themselves haven't done much, almost all the openAPI stuff has been community-contributed, and it's painful maintaining those list filters/create attrs/update attrs 😁

**Comment by JohnVillalovos** [comment]
My main idea on this is to parse it so we can come up with a list of things we are missing. It would be run periodically to assist us. Not something that runs in the CI,.

But yeah it would be great if Gitlab did get the OpenAPI spec working, but it seems like they aren't assigning resources to work on it.

**Comment by nejch** [comment]
That's true, there are a lot of endpoints that often get added silently without any mention in the release posts and we miss them.

**Comment by JohnVillalovos** [comment]
I'm going to close this as I think it is quite a major task.
